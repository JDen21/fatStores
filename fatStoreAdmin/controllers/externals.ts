export * as serviceTypes from "../services.d.ts";
export * as appConfigs from "../configs.ts";
export { default as ControllerError } from "../ControllerError.ts";
export { default as ServiceError } from "../ServiceError.ts";
export * as commonTypes from "../types.d.ts";
