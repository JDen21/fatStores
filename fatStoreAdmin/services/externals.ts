export * as serviceTypes from "../services.d.ts";
export * as appConfigs from "../configs.ts";
export { CreatableRoles, Roles } from "../configs.ts";

export { safeAsync } from "../utils.ts";
export { default as ServiceError } from "../ServiceError.ts";
export type { possibleErrorTypes } from "../types.d.ts";

type methods = string;
type service = Record<string, methods[]>;
type role = string;
type permissionsList = Record<
  role,
  "*" | service | undefined
>;
import rp from "../rolePerms.default.json" with { type: "json" };
export const rolePermissions = rp as permissionsList;
