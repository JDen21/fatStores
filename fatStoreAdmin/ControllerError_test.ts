import { ControllerError } from "./controllers/externals.ts";
import { assertSnapshot } from "@std/testing/snapshot";
import { assertEquals } from "@std/assert";

const runSnapTest = Deno.args.includes("--snap") === false;

Deno.test({
  name: "ControllerError snap",
  ignore: runSnapTest,
  fn: async function (t) {
    const code = 400;
    const message = "Invalid arguments.";
    const err = new ControllerError(code, message);

    assertEquals(err.message, message);
    await assertSnapshot(t, JSON.stringify(err));
  },
});
