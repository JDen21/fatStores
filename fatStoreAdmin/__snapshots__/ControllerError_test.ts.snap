export const snapshot = {};

snapshot[`ControllerError snap 1`] = `'{"code":400}'`;
